include/itkMorphologicalContourInterpolator.h

```diff
@@ -113,14 +169,15 @@ class MorphologicalContourInterpolator :public ImageToImageFilter < TImage, TIma
   bool                       m_HeuristicAlignment;
   bool                       m_UseDistanceTransform;
   bool                       m_UseBallStructuringElement;
+  bool                       m_UseCustomSlicePositions;
   bool                       m_StopSpawning; //stop spawning new threads
   IdentifierType             m_MinAlignIters; //minimum number of iterations in align method
   IdentifierType             m_MaxAlignIters; //maximum number of iterations in align method
   ::ThreadPool*              m_ThreadPool; //avoid name conflict
 
-  //grafted input and output to prevent unnecessary pipeline modification checks
-  typename TImage::Pointer m_Input;
-  typename TImage::Pointer m_Output;
+  /** Are these two slices equal? */
+  bool ImagesEqual(typename BoolSliceType::Pointer a, typename BoolSliceType::Pointer b);
```

Comment:
A best practice that may have performance implications -- using a reference in the method arguments will avoid SmartPointer creation / destruction, so

```
bool ImagesEqual(typename BoolSliceType::Pointer & a, typename BoolSliceType::Pointer & b)
```